src/util.rs

```diff
@@ -7,3 +7,82 @@ macro_rules! regex {
         &REGEX
     }};
 }
+
+use crate::char_len;
+use std::{collections::HashMap, iter::FromIterator};
+
+pub struct TranslationTable {
+    table: HashMap<char, char>,
+}
+impl TranslationTable {
+    pub fn new(f: &str, t: &str) -> Result<Self, &'static str> {
```

Comment:
Could we make this `const fn` and use `&'static str`? I think we already know this at compile time. If we used these, we could just `debug_assert_eq!(char_len(f), char_len(t))`.

Reply:
I don't think we could use `&'static str`'s because when you create the table (see below), you use refs of `self.tr_from` and `self.tr_to`.

(I have a (nearly) fully implemented version of deroff.rs that has `macro_tr` implemented, and I credited it to Anders, but I can't find where I got the code from :(, but this is from that (slightly modified to work with `TranslationTable`))
```rust
self.tr = match TranslationTable::new(&self.tr_from, &self.tr_to) {
    Ok(table) => Some(table),
    Err(e) => ..
};
```

Reply:
Yes, but looking at the code, `tr_from` and `tr_to` have essentially the same length, so in this case maybe we can just `assert!()` and let it panic since it does not have the same length?

Reply:
I mean, we _could_, but I don't quite know why that'd be preferred over a `Result`
Am I missing something?
I feel like `Result` is clearer than a comment explaining that it _can_ panic

Reply:
I think preserving a `Result` is fine for now. We can always `unwrap` directly if we want, and an invalid translation directive seems like something we should be able to handle without panicking.